frontend/components/timer/index.jsx

```diff
@@ -2,6 +2,18 @@ import React, { useEffect, useState } from "react";
 
 const Timer = ({ time = 180, onTimeIsUp }) => {
   const [seconds, setSeconds] = useState(time);
+  const [userTime, setUserTime] = useState('');
```

Comment:
What about using `createRef` instead of `useState`? That way we don't need to be updating state with every input, we can simply grab the value when we need it:
```
const userTime = createRef(null);

const handleSubmit = (e) => {
    e.preventDefault();
    const userTimeElement = userTime.current
    setSeconds(userTimeElement.value);
    userTimeElement.value = ''
};

// ....

<input ref={userTime} type="number" />
```

Reply:
Interesting implementation! I just refactored the code.